src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/AllocationService.kt

```diff
@@ -72,6 +74,12 @@ class AllocationService(
       .map { FindActiveAllocationIdsResponse(it) }
   }
 
+  fun findSuspendedAllocations(prisonId: String, excludeProgramCodes: List<String>?, courseActivityId: Long?): List<FindSuspendedAllocationsResponse> {
+    val excludePrograms = excludeProgramCodes?.takeIf { it.isNotEmpty() } ?: listOf(" ") // for unknown reasons the SQL fails on Oracle with an empty list or a zero length string
+    return findPrisonOrThrow(prisonId)
+      .let { offenderProgramProfileRepository.findSuspendedAllocations(prisonId, excludePrograms, courseActivityId, PageRequest.of(0, 100)) }
+  }
```

Comment:
Although this is very similar to the method above `findActiveAllocations` I couldn't just add another optional filter on suspended flag because a) we don't want any paging on our new endpoints and b) we need different outputs from the SQL.

Reply:
what is the reason for the `100` in the page request? Is it because you will never get more than a 100 in a prison?

Reply:
Good question - I meant to remove this.